### ISSUE TITLE
PretrainedBertEmbedder get different embedding every time

### ISSUE THREAD

**Issue by playma** [allenai/allennlp#2943]
Is it correct that I get the different embedding every time?

```
bert_embedder = PretrainedBertEmbedder(
    pretrained_model='bert-base-chinese',
    requires_grad=False,
    # top_layer_only=True  # conserve memory
)
word_embeddings = BasicTextFieldEmbedder(
    {"tokens": bert_embedder},
    allow_unmatched_keys=True
)
embedded_term1 = word_embeddings(term1)
```

**Comment by matt-gardner** [comment]
This is surprising to me.  We're using huggingface's code for pretrained BERT, so one sanity check is to try a similar test just using their code (and not allennlp).  If you still see the issue there, then it's not something we can fix.

**Comment by playma** [comment]
I found the problem.
It should add `.eval()` to fix the pretained parameters.

```
bert_embedder = PretrainedBertEmbedder(
    pretrained_model='bert-base-chinese',
    requires_grad=False,
    # top_layer_only=True  # conserve memory
).eval()
```

**Comment by matt-gardner** [comment]
Ah, right, it's probably dropout that you're seeing, not any changing parameters.  It's easy to forget about that...

I'm closing this issue; feel free to comment again if you have more questions.

**Comment by playma** [comment]
@matt-gardner 
I face a new problem.
I use the `lstm.eval()`, I still get the different `encoded_term1` everytime.
Do you have any idea? 

```
      EMBEDDING_DIM = word_embeddings.get_output_dim()
      HIDDEN_DIM = 256
      lstm = PytorchSeq2VecWrapper(torch.nn.LSTM(EMBEDDING_DIM, HIDDEN_DIM, batch_first=True))
      lstm.eval()  # Not work

      if cuda_device >= 0:
           word_embeddings = word_embeddings.cuda(cuda_device)
           lstm = lstm.cuda(cuda_device)

      embedded_term1 = word_embeddings(term1)
      term1_mask = get_text_field_mask(term1)
      encoded_term1 = lstm(embedded_term1, term1_mask)
```

**Comment by matt-gardner** [comment]
How exactly are you testing this?  That code is going to randomly initialize the LSTM.  If you're just running your script over and over, it will definitely produce different results each time.

**Comment by playma** [comment]
@mahnerak Sorry, I did a stupid job, I should not use LSTM to extract features.
